pkg/kubelet/images/image_manager.go

```diff
@@ -134,6 +134,8 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 	if !shouldPullImage(container, present) {
 		if present {
 			msg := fmt.Sprintf("Container image %q already present on machine", container.Image)
+			m.podPullingTimeRecorder.RecordImageStartedPulling(pod.UID)
+			m.podPullingTimeRecorder.RecordImageFinishedPulling(pod.UID)
```

Comment:
Users may think this is a quick pulling, however, no pulling happened.

Is this by design?

Reply:
If pod has multi-containers, we should add logic in RecordImageFinishedPulling like below:
```
diff --git a/pkg/kubelet/util/pod_startup_latency_tracker.go b/pkg/kubelet/util/pod_startup_latency_tracker.go
index 88a1b506f71..0376c67a4a8 100644
--- a/pkg/kubelet/util/pod_startup_latency_tracker.go
+++ b/pkg/kubelet/util/pod_startup_latency_tracker.go
@@ -136,7 +136,9 @@ func (p *basicPodStartupLatencyTracker) RecordImageFinishedPulling(podUID types.
                return
        }
 
-       state.lastFinishedPulling = p.clock.Now() // Now is always grater than values from the past.
+       if state.firstStartedPulling.IsZero() {
+               state.lastFinishedPulling = p.clock.Now() // Now is always greater than values from the past.
+       }
 }
```

Reply:
No it isn't by design, it was originally suggested by @aojea in the issue #114903

Reply:
I think the idea was to keep some consistency. Having the firstStartedPulling equals to lastFinishedPulling equals to Time.Now() made sense to me for image already present but I do agree with you it could be misunderstood and interpreted as a quick pulling when in fact nothing happened

Reply:
If an empty date will make more misunderstanding, this is OK to me.